pkg/controller/vsphere/reconciler.go

```diff
@@ -693,6 +698,28 @@ func (vm *virtualMachine) getPowerState() (types.VirtualMachinePowerState, error
 	}
 }
 
+// reconcileTags ensures that the required tags are present on the virtual machine, eg the Cluster ID
```

Comment:
can we elaborate what is "the required tags"? e.g this is used by the installer on cluster deletion to ensure there's no leaked resources.